kt-android-runtime/runtime/src/main/java/io/sdkgen/runtime/SdkgenHttpClient.kt

```diff
@@ -0,0 +1,229 @@
+package io.sdkgen.runtime
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Point
+import android.os.Build
+import android.provider.Settings
+import android.view.WindowManager
+import com.google.gson.*
+import okhttp3.*
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.RequestBody.Companion.toRequestBody
+import java.net.SocketTimeoutException
+import java.util.*
+import java.util.concurrent.TimeUnit
+import kotlin.concurrent.timerTask
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+@Suppress("unused")
+class SdkGenHttpClient(
+    private val baseUrl: String,
+    private val applicationContext: Context,
+    private val defaultTimeoutMillis: Long = 10000L
+) {
+
+    data class InternalResposne(val error: JsonObject?, val result: JsonObject?)
+
+    private val random = Random()
+    private val hexArray = "0123456789abcdef".toCharArray()
+    private val gson = Gson()
+    private val connectionPool = ConnectionPool(100, 100, TimeUnit.SECONDS)
+    private val httpClient = OkHttpClient.Builder()
+        .connectionPool(connectionPool)
+        .dispatcher(Dispatcher().apply { maxRequests = 200 ; maxRequestsPerHost = 200 })
+        .connectTimeout(180, TimeUnit.SECONDS)
+        .readTimeout(180, TimeUnit.SECONDS)
+        .callTimeout(180, TimeUnit.SECONDS)
+        .writeTimeout(180, TimeUnit.SECONDS)
+        .build()
+
+    private fun callId(): String {
+        val bytes = ByteArray(8)
+        random.nextBytes(bytes)
+        return bytesToHex(bytes)
+    }
+
+    private fun bytesToHex(bytes: ByteArray): String {
+        val hexChars = CharArray(bytes.size * 2)
+        for (j in bytes.indices) {
+            val v = bytes[j].toInt() and 0xFF
+            hexChars[j * 2] = hexArray[v ushr 4 ]
+            hexChars[j * 2 + 1] = hexArray[v and 0x0F]
+        }
+        return String(hexChars)
+    }
+
+    private fun language(): String {
+        val loc = Locale.getDefault()
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            return loc.toLanguageTag()
+        }
```

Comment:
We could target >=21 and remove all the code below

Reply:
the cost of keeping it and allowing a lower target seems acceptable. For now...